src/connector.rs

```diff
@@ -115,9 +119,11 @@ impl EthConnectorContract {
 
         let msg = hex::decode(data[1]).expect(ERR_FAILED_PARSE);
         let mut fee: [u8; 32] = Default::default();
+        // TODO: Check msg length?
```

Comment:
Good question. If the length is less than 52 bytes it will panics anymore. But if greater it will ignore the tail. For now, we don't have a concrete decision for that.